single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultSVGWriter.java

```diff
@@ -931,27 +931,23 @@ protected void insertArrowsAndLabels(String prefixId,
             points.add(new Point(feederNode.getDiagramCoordinates()));
         }
 
-        InitialValue init = initProvider.getInitialValue(feederNode);
+        List<FlowArrow> arrows = initProvider.getFlowArrows(feederNode);
         boolean arrowSymmetry = feederNode.getDirection() == BusCell.Direction.TOP || feederArrowSymmetry;
 
-        Optional<String> label1 = arrowSymmetry ? init.getLabel1() : init.getLabel2();
-        Optional<Direction> direction1 = arrowSymmetry ? init.getArrowDirection1() : init.getArrowDirection2();
-
-        Optional<String> label2 = arrowSymmetry ? init.getLabel2() : init.getLabel1();
-        Optional<Direction> direction2 = arrowSymmetry ? init.getArrowDirection2() : init.getArrowDirection1();
-
+        FlowArrow arrow1 = arrowSymmetry ? arrows.get(0) : arrows.get(1);
+        FlowArrow arrow2 = arrowSymmetry ? arrows.get(1) : arrows.get(0);
         int iArrow1 = arrowSymmetry ? 1 : 2;
         int iArrow2 = arrowSymmetry ? 2 : 1;
 
         // we draw the arrow only if value 1 is present
-        label1.ifPresent(lb ->
-                drawArrowAndLabel(prefixId, wireId, points, root, lb, init.getLabel3(), direction1, 0, iArrow1, metadata));
+        arrow1.getLeftLabel().ifPresent(lb ->
+                drawArrowAndLabel(prefixId, wireId, points, root, lb, arrow1.getRightLabel(), arrow1.getDirection(), 0, iArrow1, metadata));
 
         // we draw the arrow only if value 2 is present
```

Comment:
you can check that directly inside `drawArrowAndLabel`. And I'd prefer to change a bit that behaviour: it makes more sense to draw each label if non null, and the arrow only if the direction is present (even if no label is there).